src/main/java/frc/robot/subsystems/LimeLights.java

```diff
@@ -0,0 +1,119 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Limelight;
+import frc.robot.LimelightHelpers;
+
+import java.util.Optional;
+
+public class LimeLights extends SubsystemBase {
+
+    private static final String LL_Name_Banana = "limelight-banana";
+    private static final String LL_Name_Apple = "limelight-apple";
+    private boolean redAlliance;
+
+    private final Limelight limelight_banana;
+    private final Limelight limelight_apple;
+    private final AprilTagFieldLayout layout;
+    public LimeLights(){
+        limelight_banana = new Limelight(LL_Name_Banana);
+        limelight_apple = new Limelight(LL_Name_Apple);
+        layout = AprilTagFieldLayout.loadField(AprilTagFields.kDefaultField);
+        if(DriverStation.getAlliance().isPresent())
+            this.redAlliance = DriverStation.getAlliance().get() == DriverStation.Alliance.Red;
+    }
+
+    public Optional<LimelightHelpers.PoseEstimate> getRobotPoseEstimate(){
+        if(limelight_apple.targetIsSeen()){
+            return Optional.of(limelight_apple.getPoseEstimate());
+        }else if(limelight_banana.targetIsSeen()){
+            return Optional.of(limelight_banana.getPoseEstimate());
+        }
+        return Optional.empty();
+    }
+    public double getTimeStamp(){
```

Comment:
no, get this value as part of calling `getRobotPoseEstimate` instead. No reason for this value without